src/external/smartsheet.test.ts

```diff
@@ -0,0 +1,183 @@
+import createHttpError from 'http-errors';
+import { StatusCodes } from 'http-status-codes';
+
+import { smartsheetId, smartsheetToken } from '../config/env';
+import { IUserOuput } from '../db/models/User';
+import { SubscriptionStatus } from '../utils/newsletter';
+import { fetchSubscription, handleNewsletterUpdate, subscribeNewsletter, unsubscribeNewsletter } from './smartsheet';
+import * as smartsheetModule from './smartsheet';
+
+describe('Newsletter Functions', () => {
+    const user = {
+        newsletter_email: 'test@example.com',
+        first_name: 'John',
+        last_name: 'Doe',
+        affiliation: 'Health Center',
+        newsletter_subscription_status: SubscriptionStatus.SUBSCRIBED,
+        roles: ['developer'],
+    } as IUserOuput;
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('handleNewsletterUpdate', () => {
+        it('should call subscribeNewsletter when user is subscribed', async () => {
+            jest.spyOn(smartsheetModule, 'subscribeNewsletter').mockImplementationOnce(jest.fn());
+            await handleNewsletterUpdate(user);
+            expect(subscribeNewsletter).toHaveBeenCalledWith(user);
```

Comment:
Does it mean that on each user update (even if subscription status didn't change) we will call subscribeNewsletter function ?